src/main/java/org/hydev/mcpm/utils/GeneralUtils.java

```diff
@@ -41,6 +42,33 @@ public static URI makeUrl(String url, Object... parameters)
         }
     }
 
+    /**
+     * Concatenate two URLs
+     *
+     * @param base Base url
+     * @param path Path
+     * @return Concatenated URLs
+     */
+    public static URI concatUri(String base, String path)
+    {
+        // Remove redundant slashes
+        base = StringUtils.stripEnd(base, "/");
+        path = StringUtils.stripStart(path, "/");
+        return URI.create(base + "/" + path);
```

Comment:
I'm crying that you didn't find a better solution than I did. I looked everywhere... I was floored that no-one provided like a good "url resolve me" method that didn't have some weird caveat T_T